nix/commands/export-patches.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -e
+
+# wrapper fn to check if a key exists in a hashmap
+function array_key_exists() {
+  local _array_name="$1"
+  local _key="$2"
+  local _cmd='echo ${!'$_array_name'[@]}'
+  local _array_keys=($(eval $_cmd))
+  local _key_exists=$(echo " ${_array_keys[@]} " | grep " $_key " &>/dev/null; echo $?)
+  [[ "$_key_exists" = "0" ]] && return 0 || return 1
+}
+
+# init array of valid patch directories
+declare -A patch_dirs
+patch_dirs["node"]="third_party/electron_node"
+patch_dirs["v8"]="v8"
+patch_dirs["chromium"]=""
+patch_dirs["boringssl"]="third_party/boringssl"
+
+basedir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
```

Comment:
Just to match the other files can this line and source the `__load-config` file be moved to the top of this one.  Just under `set -e`